crates/sui-sdk-types/src/types/transaction/unresolved.rs

```diff
@@ -321,3 +354,27 @@ impl From<Value> for serde_json::Value {
         }
     }
 }
+
+impl From<&Object> for Input {
+    fn from(object: &Object) -> Self {
```

Comment:
This should call `object.as_input()` and `as_input`'s body should contain this logic.